src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -65,11 +67,15 @@ def run_python_test(transitive_hydrated_target, pytest):
   output_pytest_requirements_pex_filename = 'pytest-with-requirements.pex'
   requirements_pex_argv = [
     './{}'.format(pex_snapshot.files[0]),
-    '--python', python_binary,
+    '--python', text_type(python_binary),
     '-e', 'pytest:main',
     '-o', output_pytest_requirements_pex_filename,
     # Sort all user requirement strings to increase the chance of cache hits across invocations.
-  ] + list(pytest.get_requirement_strings()) + sorted(all_requirements)
+  ] + list(
+    text_type(req) for req in sorted(
+      list(pytest.get_requirement_strings())
+      + list(all_requirements))
+  )
```

Comment:
Can change this outer part to a list comp.

Reply:
Also does this need a TODO to drop after PY2?

Reply:
Yes it does!

Reply:
> Can change this outer part to a list comp.

Sorry, what does this mean exactly? I'm using `list()` twice right here because I need to be able to concatenate them together, not sure if I see a way to bring that outside.

Reply:
```python
[text_type(req) for req in sorted(list(pytest.get_requirement_strings()) + list(all_requirements))]
```

Reply:
Great! Done!